### DISCUSSION TITLE
Icon Request: table-row-insert-x, table-row-delete, table-column-insert-x, table-column-delete

### DISCUSSION THREAD

**Discussion by rozerwillis** [FortAwesome/Font-Awesome#21115] [Icon Requests]
Icon Request: fa-insert-row
Request for insert a row in table...Regards Rozer

**Comment by tagliala** [comment]
please provide single color examples suitable for 14px icon size for this request, thanks

**Comment by dfmiller** [comment]
https://forums.zoho.com/viewImage.do?fileId=2266000002474117&forumGroupId=2266000000002001

**Comment by mshovelt** [comment]
+1 Would be great to get a complete set of table editing icons: create/delete table, row insert above/below, column insert left-of/right-of, delete row, delete column.

**Comment by jpzwarte** [comment]
Some inspiration
<img width="664" alt="Screen Shot 2019-09-26 at 13 02 08" src="https://user-images.githubusercontent.com/3968/65683632-052e9f80-e05e-11e9-9e93-4d47df418e7c.png">

In order:

1. Add row above
2. Add row below
3. Delete row
4. Cut row
5. Copy row
6. Past row
7. Cut column
8. Copy column
9. Past column
10. Add column before
11. Add column after
12. Delete column
13. Merge cells
14. Split cells
15. Header row
16. Header column
17. Numbered table
18. Fill color (not relevant here)
19. Delete table

**Comment by lucianholt97** [comment]
+1 ...any updates on this issue?

**Comment by daprezjer** [comment]
+1

**Comment by 1stKarkan** [comment]
I've stumbled across several in-browser HTML-Editors and settled on tip tap. Tip tap is a **renderless**  text editor, which great because developers can fully customize it to their needs. The downside is the lack of icons. As more and more components like HTML editors are renderless, we need to add our own icons for table editing. Fontawesome really should implement such icons.

Please make this a thing

**Comment by gregveres** [comment]
+1! I was shocked that my pro subscription didn't have these icons given some of the other icons that are included that have such limited use.

**Comment by gregveres** [comment]
Hey guys, I just checked out the leader board and the problem is that there have been so many requests for table icons  spread across multiple github issues that this particular issue isn't making the list, but if you added up all the requests I bet it would make the leader board.

**Comment by gregveres** [comment]
We have:
#13814
#4555
#4554
#4553
#4550
#4237 - this is database related but same icons and was closed for this issue
#2859
#4028


There seems like a lot of interest in these icons dating back to at least 2014

**Comment by gregveres** [comment]
I think it might be time to give up on font awesome and go with the free MDI icons. They have all the table icons. I really don't want to mix and match icon styles. https://materialdesignicons.com/

**Comment by rizen** [comment]
More inspiration
![images](https://user-images.githubusercontent.com/170009/149649789-600e816d-deda-4ee9-84be-c758be7074b6.jpeg)
![row-icon-19](https://user-images.githubusercontent.com/170009/149649791-9cd98062-8d10-46cf-9886-c1e1d6b0d079.png)
![Unknown](https://user-images.githubusercontent.com/170009/149649792-c70ed20c-bc17-46e1-97c3-17732634871f.jpeg)
![632888](https://user-images.githubusercontent.com/170009/149649793-5e0e47d8-dd6f-4b11-ab57-90bfd6c7c64b.png)

**Comment by gregveres** [comment]
Another 2 years later and still no table operation icons. :(

**Comment by DNF-SaS** [comment]
Does anybody of the font awesome team look at this issue? We have also licensed pro and are surprised that no icons for any table-operations are available.

**Comment by jpzwarte** [comment]
Perhaps once @claviska starts making a rich text editor for shoelace, we will get these ;)

**Comment by chrisjowen** [comment]
+1

**Comment by cloakedninjas** [comment]
Wow, this issue is almost 10 years old... Some good icons that Confluence / JIRA use:

<img width="60" alt="Screenshot 2023-10-13 at 17 01 00" src="https://github.com/FortAwesome/Font-Awesome/assets/1395013/f78728fe-79db-4c91-9fc4-af358f24e31d">
<img width="67" alt="Screenshot 2023-10-13 at 17 00 55" src="https://github.com/FortAwesome/Font-Awesome/assets/1395013/e60c0782-360d-440c-b9b2-ff7db73654dd">

**Comment by sensibleworld** [comment]
This is a quick update to let folks know that we’re refining our icon request workflow behind the scenes, and as part of that we’re going through all open requests and filing them into several new buckets to be addressed.

In this case we’ve added this issue to our new list to be created! We can’t promise any firm timelines right now, but this will help us keep track of it.

We’ll update this issue when the icon is released into the wild 🙂

**Comment by Georgevanvelzen** [comment]
I see it's been more than a year since this ten year old request. Is there going to be an icon for the table row/column add/delete? seems pretty simple and obvious given most use cases.

**Comment by sensibleworld** [comment]
Added a bunch of these options to 7.2: https://fontawesome.com/search?q=table&ip=classic&s=solid&c=text-formatting
